secondarylib/src/main/scala/com/socrata/datacoordinator/secondary/SecondaryWatcher.scala

```diff
@@ -107,12 +107,19 @@ class SecondaryWatcher[CT, CV](universe: => Managed[SecondaryWatcher.UniverseTyp
           // Prefer cookie handed-back by the exception, but since we write notes in out cookies currently that we don't
           // want to lose, let's avoid overriding those if we are not given Some other cookie string
           manifest(u).markSecondaryDatasetBroken(job, cookie.orElse(job.initialCookie))
-        case rlse@ReplayLaterSecondaryException(reason, cookie) =>
-          if (job.replayNum < maxReplays) {
-            val replayAfter = Math.min(replayWait.toSeconds * Math.log(job.replayNum + 2), maxReplayWait.toSeconds)
+        case rlse@ReplayLaterSecondaryException(reason, cookie, delay) =>
+          val (replayDelay, replayNum) =
+            delay match {
+              case Some(d) => (d, if (job.replayNum >= maxReplays) 0 else job.replayNum)
+              case None => (replayWait, job.replayNum)
+            }
+          if (replayNum < maxReplays) {
+            val replayAfter =
+              if (delay.isDefined) replayDelay.toSeconds
```

Comment:
Is this condition needed? It seems that using else condition in either case would be a bit cleaner.

Reply:
The logarithmic function is for the standard delay value dictate by secondary watcher.  I think if the delay is specified by the plugin, it should be used as is.  Again, this is a debateble issue.  Since we are not merging this to master, we probably don't need a verdict now.